components/TokenActions/index.tsx

```diff
@@ -33,11 +38,33 @@ export const TokenActions = ({
         logo: LogoTicker;
         text: string;
     }[];
+    isImported?: boolean;
 }): JSX.Element => {
+    const removePool = useStore(selectRemovePool);
     const { network } = useStore(selectWeb3Info, shallow);
     const provider = useStore(selectProvider);
     const [isOpen, setIsOpen] = useState<boolean>(false);
     const buttonStyles = 'flex cursor-pointer items-center p-2 text-sm hover:bg-theme-button-bg-hover w-full';
+
+    const removeImportedPoolFromUi = useCallback((): void => {
+        if (poolAddress) {
+            // Check for imported Pool in localStorage
+            const localStoragePoolAddresses = localStorage.getItem('importedPools');
+            const parsedImportedPools = localStoragePoolAddresses && JSON.parse(localStoragePoolAddresses);
+            const poolIndex = parsedImportedPools?.findIndex((pool: string) => pool === poolAddress);
+            if (poolIndex !== -1) {
+                parsedImportedPools.splice(poolIndex, 1);
+                localStorage.setItem('importedPools', JSON.stringify(parsedImportedPools));
+            }
```

Comment:
this feels like it should be in a util function, we should centralize any logic that does parsing (and setting) of the localStorage pools and wrap it in a try/catch too since its possible someone could write an arbitrary value there and then the UI would get in a broken state